pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -93,8 +92,17 @@ func validateFitArgs(args config.NodeResourcesFitArgs) error {
 	return allErrs.ToAggregate()
 }
 
+type fitOptionFunc func(f *Fit)
+
+// FitWithPodOverhead sets the PodOverhead featuregate for the Fit plugin
+func FitWithPodOverhead(enable bool) fitOptionFunc {
```

Comment:
Just to save the prefix `Fit`, what about having a generic struct that holds all the booleans and to which all options in this package apply to?

Then you copy over all the needed flags from that struct to the actual objects (or just make it an unnamed member). I know not all flags apply to all objects, but 🤷‍♂️

Reply:
yeah, lets have a single one per plugin directory, they will likely share most of the features.

Reply:
An easy way to do this would be to just embed a `resourceAllocationScorer` in the `Fit` struct. It would only be used for the featuregate flags, but then we don't need to add a new type just for these. wdyt?

Reply:
-1

Better have an isolated struct just for flags